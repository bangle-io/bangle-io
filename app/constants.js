export const SPLIT_SCREEN_MIN_WIDTH = parseInt(
  getComputedStyle(document.documentElement).getPropertyValue(
    '--wide-screen-min-width',
  ),
  10,
);
