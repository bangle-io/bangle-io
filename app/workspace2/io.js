import * as idb from 'idb-keyval';

export class FileOps {
  // file
  static async getFile(wsPath) {
    return idb.get(wsPath);
  }

  static async deleteFile(wsPath) {
    const file = await this.getFile(wsPath);
    if (!file) {
      throw new FileError(`File ${wsPath} does not exists`);
    }
    return idb.del(wsPath);
  }

  static async renameFile(wsPath, newWsPath) {
    const file = await this.getFile(wsPath);

    if (!file) {
      throw new FileError(`File ${wsPath} does not exists`);
    }

    const newFile = await this.getFile(newWsPath);

    if (newFile) {
      throw new FileError(`File ${newWsPath} already exists`);
    }

    await this.deleteFile(wsPath);
    return this.createFile(newWsPath, file);
  }

  static async createFile(wsPath, payload) {
    if (await this.getFile(wsPath)) {
      throw new FileError(`File ${wsPath} already exists`);
    }
    return idb.set(wsPath, payload);
  }

  static async updateFile(wsPath, payload) {
    if (!(await this.getFile(wsPath))) {
      throw new FileError(`File ${wsPath} does not exists`);
    }
    return idb.set(wsPath, payload);
  }

  static async listFiles(wsName) {
    const keys = (await idb.keys()) || [];

    return keys.filter((k) => k.startsWith(wsName + ':'));
  }
}

export class FileError extends Error {
  constructor(message) {
    super('FileError: ' + message);
    this.name = 'FileError';
  }
}
export class FilePermissionError extends Error {
  constructor(message) {
    super('FilePermissionError: ' + message);
    this.name = 'FilePermissionError';
  }
}
