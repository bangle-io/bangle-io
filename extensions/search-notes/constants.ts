import type { SearchResultItem } from '@bangle.io/search-pm-node';

export const CONCURRENCY = 10;

export const SHOW_SEARCH_SIDEBAR_OPERATION =
  'operation::@bangle.io/search-notes:show-search-sidebar';

export const SEARCH_SIDEBAR_NAME =
  'sidebar::@bangle.io/search-notes:search-notes';
export const extensionName = '@bangle.io/search-notes';
export interface SearchNotesExtensionState {
  searchQuery: string;
  pendingSearch: boolean;
  searchResults: SearchResultItem[] | null;
  // a counter keeping track of external change to the search query
  // useful for resetting any local state
  externalInputChange: number;
}

export type SearchNotesActions =
  | {
      name: `action::@bangle.io/search-notes:update-state`;
      value: Partial<SearchNotesExtensionState>;
    }
  | {
      name: 'action::@bangle.io/search-notes:input-search-query';
      value: {
        query: string;
      };
    }
  | {
      name: 'action::@bangle.io/search-notes:external-search-query-update';
      value: {
        query: string;
      };
    };
