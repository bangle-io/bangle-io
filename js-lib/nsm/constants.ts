export const CORE_ACTION_ON_READY = '$$nsm:onReady';
