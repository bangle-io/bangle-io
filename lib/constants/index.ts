// This package should not import any other package and should be mostly
// primitives.
export * from './core/core-dialogs';
export * from './core/core-operations';
export * from './core/core-palettes';
export * from './editor';
export * from './help-fs';
export * from './search';
export * from './store';
export * from './ui';
export * from './worker';
export * from './workspace';
export * from './write-disk';
