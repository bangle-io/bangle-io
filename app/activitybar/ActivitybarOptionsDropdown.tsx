import React, { useCallback } from 'react';

import { useSerialOperationContext } from '@bangle.io/api';
import { useNsmSlice } from '@bangle.io/bangle-store-context';
import {
  CHANGELOG_MODAL_NAME,
  CORE_OPERATIONS_NEW_NOTE,
  CORE_OPERATIONS_NEW_WORKSPACE,
  CORE_PALETTES_TOGGLE_NOTES_PALETTE,
  CORE_PALETTES_TOGGLE_OPERATION_PALETTE,
  CORE_PALETTES_TOGGLE_WORKSPACE_PALETTE,
  CorePalette,
} from '@bangle.io/constants';
import type { SidebarType } from '@bangle.io/extension-registry';
import type { SerialOperationKeybindingMapping } from '@bangle.io/shared-types';
import { nsmUI, nsmUISlice } from '@bangle.io/slice-ui';
import {
  BangleIcon,
  DiscordIcon,
  DotsVerticalIcon,
  DropdownMenu,
  GiftIcon,
  MenuItem,
  MenuSection,
  PrettyKeybinding,
  SettingsIcon,
  TwitterIcon,
} from '@bangle.io/ui-components';

import { ButtonStyleOBj } from './common';

export const ActionPaletteKey = 'ActionPalette';
export const DiscordKey = 'Discord';
export const NewNoteKey = 'NewNote';
export const NewWorkspaceKey = 'NewWorkspace';
export const NotesPaletteKey = 'NotesPalette';
export const ReportIssueKey = 'ReportIssue';
export const SwitchWorkspaceKey = 'SwitchWorkspace';
export const ToggleThemeKey = 'ToggleTheme';
export const TwitterKey = 'Twitter';
export const WhatsNewKey = 'WhatsNewKey';

type AllKeysType =
  | typeof ActionPaletteKey
  | typeof DiscordKey
  | typeof NewNoteKey
  | typeof NewWorkspaceKey
  | typeof NotesPaletteKey
  | typeof ReportIssueKey
  | typeof SwitchWorkspaceKey
  | typeof ToggleThemeKey
  | typeof TwitterKey
  | typeof WhatsNewKey;

export function ActivitybarOptionsDropdown({
  widescreen,
  operationKeybindings,
  sidebarItems,
  activeSidebar,
}: {
  widescreen: boolean;
  operationKeybindings: SerialOperationKeybindingMapping;
  sidebarItems?: SidebarType[];
  activeSidebar?: string;
}) {
  const [, uiDispatch] = useNsmSlice(nsmUISlice);

  const { dispatchSerialOperation } = useSerialOperationContext();

  const handleDropdown = useCallback(
    (k: any) => {
      let key: AllKeysType = k;
      switch (key) {
        case ActionPaletteKey: {
          uiDispatch(nsmUI.togglePalette(CorePalette.Operation));
          break;
        }
        case NewNoteKey: {
          dispatchSerialOperation({
            name: CORE_OPERATIONS_NEW_NOTE,
          });
          break;
        }
        case NewWorkspaceKey: {
          dispatchSerialOperation({
            name: CORE_OPERATIONS_NEW_WORKSPACE,
          });
          break;
        }
        case NotesPaletteKey: {
          uiDispatch(nsmUI.togglePalette(CorePalette.Notes));
          break;
        }
        case ReportIssueKey: {
          window.open(
            'https://github.com/bangle-io/bangle-io/issues',
            '_blank',
          );
          break;
        }
        case SwitchWorkspaceKey: {
          uiDispatch(nsmUI.togglePalette(CorePalette.Workspace));
          break;
        }
        case ToggleThemeKey: {
          uiDispatch(nsmUI.toggleColorSchema());
          break;
        }
        case DiscordKey: {
          window.open('https://discord.gg/GvvbWJrVQY', '_blank');
          break;
        }
        case TwitterKey: {
          window.open('https://twitter.com/bangle_io', '_blank');
          break;
        }
        case WhatsNewKey: {
          uiDispatch(
            nsmUI.showDialog({
              dialogName: CHANGELOG_MODAL_NAME,
            }),
          );

          break;
        }
        default: {
          const match = sidebarItems?.find((i) => i.name === k);

          if (match) {
            if (match.name !== activeSidebar) {
              uiDispatch(nsmUI.changeSidebar(match.name));
            }
          }
        }
      }
    },
    [activeSidebar, dispatchSerialOperation, uiDispatch, sidebarItems],
  );

  const sidebarChildren: any =
    sidebarItems?.map((item) => {
      return (
        <MenuItem
          key={item.name}
          textValue={item.title}
          aria-label={item.title}
        >
          {item.title}
          {React.cloneElement(item.activitybarIcon, {
            className:
              (item.activitybarIcon.props.className || '') + ' w-5 h-5',
          })}
        </MenuItem>
      );
    }) || null;

  return (
    <DropdownMenu
      buttonProps={{
        variant: 'transparent',
        ariaLabel: 'options menu',
        size: widescreen ? 'lg' : 'sm',
        tone: 'secondary',
        //  we need this to use the activitybar color of the icon
        style: widescreen ? ButtonStyleOBj.normal : undefined,

        leftIcon: widescreen ? <SettingsIcon /> : <DotsVerticalIcon />,
        onHoverStyle: ButtonStyleOBj.hover,
        onPressStyle: ButtonStyleOBj.press,
      }}
      menuProps={{
        ariaLabel: 'options dropdown',
        placement: widescreen ? 'right-start' : 'bottom-start',
      }}
      onAction={handleDropdown}
    >
      <MenuSection aria-label="misc section">
        <MenuItem aria-label="new note" key={NewNoteKey}>
          New note
        </MenuItem>
        <MenuItem aria-label="new workspace" key={NewWorkspaceKey}>
          New workspace
        </MenuItem>
        <MenuItem
          aria-label="switch workspace"
          key={SwitchWorkspaceKey}
          textValue="switch workspace"
        >
          <span>Switch workspace</span>
          {widescreen && (
            <PrettyKeybinding
              rawKey={
                operationKeybindings[CORE_PALETTES_TOGGLE_WORKSPACE_PALETTE] ||
                ''
              }
            />
          )}
        </MenuItem>
      </MenuSection>
      {sidebarChildren && (
        <MenuSection aria-label="activitybar section">
          {sidebarChildren}
        </MenuSection>
      )}
      <MenuSection aria-label="ui section">
        <MenuItem aria-label="Switch Dark/Light theme" key={ToggleThemeKey}>
          Switch Dark/Light theme
        </MenuItem>
      </MenuSection>
      <MenuSection aria-label="palettes">
        <MenuItem
          key={NotesPaletteKey}
          textValue="notes palette"
          aria-label="notes palette"
        >
          <span>Notes palette</span>
          {widescreen && (
            <PrettyKeybinding
              rawKey={
                operationKeybindings[CORE_PALETTES_TOGGLE_NOTES_PALETTE] || ''
              }
            />
          )}
        </MenuItem>
        <MenuItem
          key={ActionPaletteKey}
          textValue="operation palette"
          aria-label="operation palette"
        >
          <span>Operation palette</span>
          {widescreen && (
            <PrettyKeybinding
              rawKey={
                operationKeybindings[CORE_PALETTES_TOGGLE_OPERATION_PALETTE] ||
                ''
              }
            />
          )}
        </MenuItem>
      </MenuSection>
      <MenuSection aria-label="links section">
        <MenuItem
          key={WhatsNewKey}
          textValue="whats new"
          aria-label="whats new"
        >
          <span>Whats new</span>
          <GiftIcon className="w-5 h-5" />
        </MenuItem>
        <MenuItem
          key={ReportIssueKey}
          textValue="report issue"
          aria-label="report issue"
        >
          <span>Report issue</span>
          <BangleIcon className="w-5 h-5" />
        </MenuItem>
        <MenuItem
          key={TwitterKey}
          textValue="follow twitter"
          aria-label="follow twitter"
        >
          <span>Twitter</span>
          <TwitterIcon className="w-5 h-5" />
        </MenuItem>
        <MenuItem
          key={DiscordKey}
          textValue="join discord"
          aria-label="join discord"
        >
          <span>Discord</span>
          <DiscordIcon className="w-5 h-5" />
        </MenuItem>
      </MenuSection>
    </DropdownMenu>
  );
}
