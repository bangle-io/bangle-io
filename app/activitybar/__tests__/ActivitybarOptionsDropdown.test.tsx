/**
 * @jest-environment @bangle.io/jsdom-env
 */
import { act, fireEvent, render, waitFor } from '@testing-library/react';
import React from 'react';

import { useSerialOperationContext } from '@bangle.io/api';
import { CORE_PALETTES_TOGGLE_WORKSPACE_PALETTE } from '@bangle.io/constants';

import { ActivitybarOptionsDropdown } from '../ActivitybarOptionsDropdown';

jest.mock('@bangle.io/api', () => {
  const rest = jest.requireActual('@bangle.io/api');

  return {
    ...rest,
    useSerialOperationContext: jest.fn(() => ({})),
  };
});

jest.mock('react-dom', () => {
  const otherThings = jest.requireActual('react-dom');

  return {
    ...otherThings,
    createPortal: jest.fn((element, node) => {
      return element;
    }),
  };
});

const operationKeybindings = {
  [CORE_PALETTES_TOGGLE_WORKSPACE_PALETTE]: 'Ctrl-P',
};

let useSerialOperationContextMock =
  useSerialOperationContext as jest.MockedFunction<
    typeof useSerialOperationContext
  >;

let dispatchSerialOperationMock = jest.fn();
beforeEach(() => {
  dispatchSerialOperationMock = jest.fn();
  useSerialOperationContextMock.mockImplementation(() => {
    return { dispatchSerialOperation: dispatchSerialOperationMock };
  });
});

test('renders correctly', () => {
  let result = render(
    <div>
      <ActivitybarOptionsDropdown
        operationKeybindings={operationKeybindings}
        widescreen={true}
      />
    </div>,
  );

  expect(result.container).toMatchSnapshot();
});

test('renders mobile correctly', () => {
  let result = render(
    <div>
      <ActivitybarOptionsDropdown
        operationKeybindings={operationKeybindings}
        widescreen={false}
      />
    </div>,
  );

  expect(result.container).toMatchSnapshot();
});

test('clicking the button shows dropdown', async () => {
  let result = render(
    <div>
      <ActivitybarOptionsDropdown
        operationKeybindings={operationKeybindings}
        widescreen={true}
      />
    </div>,
  );

  act(() => {
    fireEvent.click(result.getByLabelText('options menu'));
  });

  let targetOption: ReturnType<typeof result.getByLabelText> | undefined;
  await waitFor(() => {
    targetOption = result.getByLabelText('options dropdown');
    expect(targetOption).toBeTruthy();
  });

  // check if keyboard shortcut is shown
  // note: kbd utility prettifies 'Ctrl-P' to 'Ctrl-⇧'
  expect(
    [...(targetOption?.querySelectorAll('li[data-key]') || [])].find((t) =>
      t.innerHTML.includes('⇧'),
    ),
  ).toBeTruthy();

  expect(targetOption).toMatchSnapshot();
});

test('clicking items in dropdown dispatches event', async () => {
  let result = render(
    <div>
      <ActivitybarOptionsDropdown
        operationKeybindings={operationKeybindings}
        widescreen={true}
      ></ActivitybarOptionsDropdown>
    </div>,
  );

  act(() => {
    fireEvent.click(result.getByLabelText('options menu'));
  });

  let targetOption;

  await waitFor(() => {
    targetOption = result.getByLabelText('new note');
    expect(targetOption).toBeTruthy();

    fireEvent.click(targetOption);
  });

  expect(dispatchSerialOperationMock).toBeCalledTimes(1);
});
