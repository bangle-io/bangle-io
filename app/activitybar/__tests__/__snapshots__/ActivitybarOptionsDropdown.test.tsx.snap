// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clicking the button shows dropdown 1`] = `
<ul
  aria-label="options dropdown"
  class="flex flex-col min-w-72 z-dropdown shadow-md  bg-colorBgLayerFloat border-neutral p-1 py-2 rounded "
  role="menu"
  tabindex="0"
>
  <li
    role="presentation"
  >
    <ul
      aria-label="misc section"
      class="p-0 list-outside"
      role="group"
    >
      <li
        aria-disabled="false"
        aria-label="new note"
        class="outline-none cursor-pointer text-sm rounded-md px-2 py-1 "
        data-key="NewNote"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="inline-flex justify-between w-full"
        >
          New note
        </span>
      </li>
      <li
        aria-disabled="false"
        aria-label="new workspace"
        class="outline-none cursor-pointer text-sm rounded-md px-2 py-1 "
        data-key="NewWorkspace"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="inline-flex justify-between w-full"
        >
          New workspace
        </span>
      </li>
      <li
        aria-disabled="false"
        aria-label="switch workspace"
        class="outline-none cursor-pointer text-sm rounded-md px-2 py-1 "
        data-key="SwitchWorkspace"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="inline-flex justify-between w-full"
        >
          <span>
            Switch workspace
          </span>
          <span
            class="inline B-ui-components_pretty-keybinding text-colorNeutralTextSubdued"
          >
            <kbd
              class="px-1 font-sans capitalize rounded-sm B-ui-components_pretty-keybinding-kbd"
            >
              Ctrl
            </kbd>
            <kbd
              class="px-1 font-sans capitalize rounded-sm B-ui-components_pretty-keybinding-kbd"
            >
              â‡§
            </kbd>
            <kbd
              class="px-1 font-sans capitalize rounded-sm B-ui-components_pretty-keybinding-kbd"
            >
              p
            </kbd>
          </span>
        </span>
      </li>
    </ul>
  </li>
  <li
    class="mx-1 my-2 border-t-1 border-colorNeutralBorder"
    role="separator"
  />
  <li
    role="presentation"
  >
    <ul
      aria-label="ui section"
      class="p-0 list-outside"
      role="group"
    >
      <li
        aria-disabled="false"
        aria-label="Switch Dark/Light theme"
        class="outline-none cursor-pointer text-sm rounded-md px-2 py-1 "
        data-key="ToggleTheme"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="inline-flex justify-between w-full"
        >
          Switch Dark/Light theme
        </span>
      </li>
    </ul>
  </li>
  <li
    class="mx-1 my-2 border-t-1 border-colorNeutralBorder"
    role="separator"
  />
  <li
    role="presentation"
  >
    <ul
      aria-label="palettes"
      class="p-0 list-outside"
      role="group"
    >
      <li
        aria-disabled="false"
        aria-label="notes palette"
        class="outline-none cursor-pointer text-sm rounded-md px-2 py-1 "
        data-key="NotesPalette"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="inline-flex justify-between w-full"
        >
          <span>
            Notes palette
          </span>
          <span
            class="inline B-ui-components_pretty-keybinding text-colorNeutralTextSubdued"
          >
            <kbd
              class="px-1 font-sans capitalize rounded-sm B-ui-components_pretty-keybinding-kbd"
            />
          </span>
        </span>
      </li>
      <li
        aria-disabled="false"
        aria-label="operation palette"
        class="outline-none cursor-pointer text-sm rounded-md px-2 py-1 "
        data-key="ActionPalette"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="inline-flex justify-between w-full"
        >
          <span>
            Operation palette
          </span>
          <span
            class="inline B-ui-components_pretty-keybinding text-colorNeutralTextSubdued"
          >
            <kbd
              class="px-1 font-sans capitalize rounded-sm B-ui-components_pretty-keybinding-kbd"
            />
          </span>
        </span>
      </li>
    </ul>
  </li>
  <li
    class="mx-1 my-2 border-t-1 border-colorNeutralBorder"
    role="separator"
  />
  <li
    role="presentation"
  >
    <ul
      aria-label="links section"
      class="p-0 list-outside"
      role="group"
    >
      <li
        aria-disabled="false"
        aria-label="whats new"
        class="outline-none cursor-pointer text-sm rounded-md px-2 py-1 "
        data-key="WhatsNewKey"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="inline-flex justify-between w-full"
        >
          <span>
            Whats new
          </span>
          <svg
            class="w-5 h-5"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8v13m0-13V6a2 2 0 112 2h-2zm0 0V5.5A2.5 2.5 0 109.5 8H12zm-7 4h14M5 12a2 2 0 110-4h14a2 2 0 110 4M5 12v7a2 2 0 002 2h10a2 2 0 002-2v-7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </span>
      </li>
      <li
        aria-disabled="false"
        aria-label="report issue"
        class="outline-none cursor-pointer text-sm rounded-md px-2 py-1 "
        data-key="ReportIssue"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="inline-flex justify-between w-full"
        >
          <span>
            Report issue
          </span>
          <svg
            class="w-5 h-5"
            fill="none"
            viewBox="0 0 320 320"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M237.44 235.98c24.29-24.72 40.6-53.21 47.18-79.5 6.54-26.17 3.59-51.14-12.26-67.28-15.86-16.14-40.4-19.13-66.1-12.48-25.84 6.7-53.84 23.3-78.12 48.01-24.29 24.72-40.6 53.21-47.17 79.5-6.55 26.18-3.6 51.14 12.26 67.28 15.85 16.14 40.39 19.13 66.1 12.48 25.83-6.7 53.83-23.3 78.11-48.01Z"
              stroke="#000"
              stroke-width="10"
            />
            <path
              d="M230.93 231.25c24.29-24.72 40.6-53.22 47.17-79.5 6.55-26.18 3.6-51.14-12.26-67.28-15.85-16.14-40.38-19.14-66.1-12.48-25.83 6.69-53.83 23.3-78.11 48-24.29 24.72-40.6 53.22-47.17 79.51-6.55 26.17-3.6 51.14 12.26 67.27 15.85 16.14 40.38 19.14 66.1 12.48 25.83-6.69 53.83-23.29 78.11-48Z"
              stroke="#000"
              stroke-width="10"
            />
            <path
              d="M225.35 226.51c24.29-24.71 40.6-53.21 47.17-79.5 6.54-26.17 3.6-51.14-12.26-67.28-15.85-16.13-40.39-19.13-66.1-12.47-25.83 6.69-53.83 23.29-78.12 48-24.28 24.72-40.6 53.22-47.17 79.5-6.54 26.18-3.6 51.14 12.26 67.28s40.4 19.14 66.1 12.48c25.84-6.69 53.84-23.3 78.12-48Z"
              stroke="#000"
              stroke-width="10"
            />
            <path
              d="M223.5 224.62c24.28-24.72 40.59-53.21 47.16-79.5 6.54-26.18 3.6-51.14-12.26-67.28-15.85-16.14-40.39-19.14-66.1-12.48-25.83 6.7-53.83 23.3-78.12 48.01-24.28 24.72-40.6 53.21-47.17 79.5-6.54 26.17-3.6 51.14 12.26 67.28s40.4 19.13 66.1 12.48c25.84-6.7 53.84-23.3 78.12-48.01Z"
              stroke="#7FCEEE"
              stroke-width="10"
            />
            <path
              d="M216.98 219.88c24.28-24.71 40.6-53.2 47.17-79.5 6.54-26.17 3.6-51.14-12.26-67.27-15.86-16.14-40.39-19.14-66.1-12.48-25.83 6.69-53.83 23.3-78.12 48-24.28 24.72-40.6 53.22-47.17 79.51-6.54 26.17-3.6 51.14 12.26 67.27 15.86 16.14 40.39 19.14 66.1 12.48 25.84-6.69 53.83-23.29 78.12-48Z"
              stroke="#7FCEEE"
              stroke-width="10"
            />
            <path
              d="M213.26 216.1c24.28-24.72 40.6-53.22 47.17-79.5 6.54-26.18 3.6-51.14-12.26-67.28s-40.39-19.14-66.1-12.48c-25.83 6.7-53.83 23.3-78.12 48-24.28 24.72-40.6 53.22-47.17 79.51-6.54 26.17-3.6 51.14 12.26 67.28 15.86 16.13 40.39 19.13 66.1 12.48 25.84-6.7 53.83-23.3 78.12-48.01Z"
              stroke="#7FCEEE"
              stroke-width="10"
            />
            <path
              d="M207.68 211.36c24.28-24.71 40.6-53.2 47.17-79.5 6.54-26.17 3.6-51.14-12.26-67.27-15.86-16.14-40.4-19.14-66.1-12.48-25.84 6.69-53.84 23.29-78.12 48-24.29 24.72-40.6 53.22-47.17 79.5-6.54 26.18-3.6 51.15 12.26 67.28 15.86 16.14 40.39 19.14 66.1 12.48 25.83-6.69 53.83-23.3 78.12-48Z"
              stroke="#7FCEEE"
              stroke-width="10"
            />
            <path
              d="M203.96 207.58c24.28-24.72 40.6-53.22 47.17-79.5 6.54-26.18 3.6-51.14-12.26-67.28s-40.4-19.14-66.1-12.48c-25.84 6.69-53.84 23.3-78.12 48-24.29 24.72-40.6 53.22-47.17 79.51-6.54 26.17-3.6 51.14 12.26 67.28 15.85 16.13 40.39 19.13 66.1 12.47 25.83-6.69 53.83-23.29 78.12-48Z"
              stroke="#7FCEEE"
              stroke-width="10"
            />
            <path
              d="M200.24 203.79c24.28-24.72 40.6-53.21 47.16-79.5 6.55-26.17 3.6-51.14-12.25-67.28-15.86-16.14-40.4-19.13-66.1-12.48-25.84 6.7-53.84 23.3-78.12 48.01-24.29 24.72-40.6 53.21-47.17 79.5-6.55 26.18-3.6 51.14 12.26 67.28 15.85 16.14 40.38 19.13 66.1 12.48 25.83-6.7 53.83-23.3 78.12-48.01Z"
              stroke="#000"
              stroke-width="10"
            />
            <path
              d="M196.51 200c24.29-24.71 40.6-53.21 47.17-79.5 6.55-26.17 3.6-51.14-12.26-67.28-15.85-16.13-40.38-19.13-66.1-12.47-25.83 6.69-53.83 23.29-78.11 48-24.29 24.72-40.6 53.22-47.17 79.5-6.55 26.18-3.6 51.15 12.26 67.28 15.85 16.14 40.38 19.14 66.1 12.48 25.83-6.69 53.83-23.3 78.11-48Z"
              stroke="#000"
              stroke-width="10"
            />
            <path
              d="M191.86 195.27c24.29-24.72 40.6-53.21 47.17-79.5 6.54-26.17 3.6-51.14-12.26-67.28-15.85-16.14-40.39-19.14-66.1-12.48-25.83 6.7-53.83 23.3-78.12 48.01-24.28 24.72-40.6 53.21-47.17 79.5-6.54 26.17-3.59 51.14 12.27 67.28 15.85 16.13 40.38 19.13 66.1 12.48 25.83-6.7 53.83-23.3 78.11-48.01Z"
              stroke="#000"
              stroke-width="10"
            />
          </svg>
        </span>
      </li>
      <li
        aria-disabled="false"
        aria-label="follow twitter"
        class="outline-none cursor-pointer text-sm rounded-md px-2 py-1 "
        data-key="Twitter"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="inline-flex justify-between w-full"
        >
          <span>
            Twitter
          </span>
          <svg
            class="w-5 h-5"
            fill="#1DA1F2"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.29 20.251c7.547 0 11.675-6.253 11.675-11.675 0-.178 0-.355-.012-.53A8.348 8.348 0 0022 5.92a8.19 8.19 0 01-2.357.646 4.118 4.118 0 001.804-2.27 8.224 8.224 0 01-2.605.996 4.107 4.107 0 00-6.993 3.743 11.65 11.65 0 01-8.457-4.287 4.106 4.106 0 001.27 5.477A4.072 4.072 0 012.8 9.713v.052a4.105 4.105 0 003.292 4.022 4.095 4.095 0 01-1.853.07 4.108 4.108 0 003.834 2.85A8.233 8.233 0 012 18.407a11.616 11.616 0 006.29 1.84"
            />
          </svg>
        </span>
      </li>
      <li
        aria-disabled="false"
        aria-label="join discord"
        class="outline-none cursor-pointer text-sm rounded-md px-2 py-1 "
        data-key="Discord"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="inline-flex justify-between w-full"
        >
          <span>
            Discord
          </span>
          <svg
            class="w-5 h-5"
            fill="currentColor"
            viewBox="0 0 71 55"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#a)"
            >
              <path
                d="M60.1 4.9A58.55 58.55 0 0 0 45.65.42a.22.22 0 0 0-.23.1 40.78 40.78 0 0 0-1.8 3.7 54.05 54.05 0 0 0-16.23 0 37.4 37.4 0 0 0-1.83-3.7.23.23 0 0 0-.23-.1c-5.07.87-9.92 2.4-14.45 4.48a.2.2 0 0 0-.1.08C1.58 18.73-.94 32.14.3 45.39c0 .07.05.13.1.17a58.88 58.88 0 0 0 17.72 8.96c.1.03.2 0 .25-.08a42.08 42.08 0 0 0 3.63-5.9.22.22 0 0 0-.12-.31 38.77 38.77 0 0 1-5.54-2.64.23.23 0 0 1-.02-.38l1.1-.86a.22.22 0 0 1 .23-.03 41.99 41.99 0 0 0 35.68 0 .22.22 0 0 1 .23.02c.36.3.73.59 1.1.87.13.1.12.3-.02.38a36.38 36.38 0 0 1-5.54 2.63.23.23 0 0 0-.12.32 47.25 47.25 0 0 0 3.63 5.9c.05.07.15.1.24.08 5.8-1.8 11.69-4.5 17.76-8.96.06-.04.09-.1.1-.17C72.16 30.08 68.2 16.78 60.2 5a.18.18 0 0 0-.1-.1ZM23.73 37.33c-3.5 0-6.38-3.22-6.38-7.16 0-3.95 2.82-7.16 6.38-7.16 3.58 0 6.43 3.24 6.38 7.16 0 3.94-2.83 7.16-6.38 7.16Zm23.59 0c-3.5 0-6.38-3.22-6.38-7.16 0-3.95 2.82-7.16 6.38-7.16 3.58 0 6.43 3.24 6.38 7.16 0 3.94-2.8 7.16-6.38 7.16Z"
                fill="#5865F2"
              />
            </g>
          </svg>
        </span>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`renders correctly 1`] = `
<div>
  <div>
    <button
      aria-label="options menu"
      aria-pressed="false"
      class="text-lg font-600 h-11 min-w-12 px-4  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
      style="background-color: transparent; border-radius: 0; padding: 0px;"
      type="button"
    >
      <span
        class="flex flex-grow-1 overflow-hidden "
        style="justify-content: center;"
      >
        <span>
          <svg
            class="w-7 h-7  "
            fill="currentColor"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="#ff13dc"
              height="18"
              opacity="0"
              width="18"
            />
            <path
              d="M16.45,7.8965H14.8945a5.97644,5.97644,0,0,0-.921-2.2535L15.076,4.54a.55.55,0,0,0,.00219-.77781L15.076,3.76l-.8365-.836a.55.55,0,0,0-.77781-.00219L13.4595,2.924,12.357,4.0265a5.96235,5.96235,0,0,0-2.2535-.9205V1.55a.55.55,0,0,0-.55-.55H8.45a.55.55,0,0,0-.55.55V3.106a5.96235,5.96235,0,0,0-2.2535.9205l-1.1-1.1025a.55.55,0,0,0-.77781-.00219L3.7665,2.924,2.924,3.76a.55.55,0,0,0-.00219.77781L2.924,4.54,4.0265,5.643a5.97644,5.97644,0,0,0-.921,2.2535H1.55a.55.55,0,0,0-.55.55V9.55a.55.55,0,0,0,.55.55H3.1055a5.967,5.967,0,0,0,.921,2.2535L2.924,13.4595a.55.55,0,0,0-.00219.77782l.00219.00218.8365.8365a.55.55,0,0,0,.77781.00219L4.5405,15.076,5.643,13.9735a5.96235,5.96235,0,0,0,2.2535.9205V16.45a.55.55,0,0,0,.55.55H9.55a.55.55,0,0,0,.55-.55V14.894a5.96235,5.96235,0,0,0,2.2535-.9205L13.456,15.076a.55.55,0,0,0,.77782.00219L14.236,15.076l.8365-.8365a.55.55,0,0,0,.00219-.77781l-.00219-.00219L13.97,12.357a5.967,5.967,0,0,0,.921-2.2535H16.45a.55.55,0,0,0,.55-.55V8.45a.55.55,0,0,0-.54649-.55349ZM11.207,9A2.207,2.207,0,1,1,9,6.793H9A2.207,2.207,0,0,1,11.207,9Z"
            />
          </svg>
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`renders mobile correctly 1`] = `
<div>
  <div>
    <button
      aria-label="options menu"
      aria-pressed="false"
      class="text-sm font-600 h-8 min-w-8 px-2  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
      style="background-color: transparent;"
      type="button"
    >
      <span
        class="flex flex-grow-1 overflow-hidden "
        style="justify-content: center;"
      >
        <span>
          <svg
            class="w-5 h-5  "
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 6a2 2 0 110-4 2 2 0 010 4zM10 12a2 2 0 110-4 2 2 0 010 4zM10 18a2 2 0 110-4 2 2 0 010 4z"
            />
          </svg>
        </span>
      </span>
    </button>
  </div>
</div>
`;
